README.adoc

```diff
@@ -1,43 +1,186 @@
+  kn-source-kamelet describe-type NAME -o yaml
+
+Flags:
+      --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
+  -h, --help                          help for describe-type
+  -n, --namespace string              Specify the namespace to operate in.
+  -o, --output string                 Output format. One of: json|yaml|name|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file|url.
+      --show-managed-fields           If true, keep the managedFields when printing objects in JSON or YAML format.
+      --template string               Template string or path to template file to use when -o=go-template, -o=go-template-file. The template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].
+  -v, --verbose                       More output.
+----
+
+=== `binding`
+
+----
+Configure and manage a Kamelet binding.
+
+Usage:
+  kn-source-kamelet binding [command]
+
+Examples:
+
+  # Configure and manage a Kamelet binding.
+  kn-source-kamelet binding create|update|delete
+
+Available Commands:
+  create      Create Kamelet bindings and bind source to Knative broker, channel or service.
+
+Flags:
+  -h, --help   help for binding
+
+Use "kn-source-kamelet binding [command] --help" for more information about a command.
+----
+
+==== `binding create`
+
+----
+Create Kamelet bindings and bind source to Knative broker, channel or service.
+
+Usage:
+  kn-source-kamelet binding create [flags]
+
+Examples:
+
+  # Create Kamelet binding with source and sink.
+  kn-source-kamelet binding create NAME
+
+  # Add a binding properties
+  kn-source-kamelet binding create NAME --kamelet=name --sink|broker|channel|service=<name> --source-property=<key>=<value> --sink-property=<key>=<value>
+
+Flags:
+      --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
+      --broker string                 Uses a broker as binding sink.
+      --channel string                Uses a channel as binding sink.
+  -h, --help                          help for create
+      --kamelet string                Kamelet source.
+  -n, --namespace string              Specify the namespace to operate in.
+  -o, --output string                 Output format. One of: json|yaml|name|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file|url.
+      --service string                Uses a Knative service as binding sink.
+      --show-managed-fields           If true, keep the managedFields when printing objects in JSON or YAML format.
+      --sink string                   Sink expression to define the binding sink.
+      --sink-property stringArray     Add a sink property in the form of "<key>=<value>"
```

Comment:
What actually is a `sink-property` ? The `source-property` is one of the properties that you see with `kn source kamelet describe-type`, but what is the `sink-property` for ?

Reply:
ah, this is when the sink is another Kamelet. as we currently do not support sinks of type Kamelet in this plugin we can safely remove the sink property.